GAE/src/org/akvo/flow/domain/DataUtils.java

```diff
@@ -106,15 +103,16 @@ public static String jsonResponsesToPipeSeparated(String optionResponses) {
     }
 
     /**
-     * Convert a JSON string response to a list containing corresponding maps
+     * Convert a JSON string response to a list of corresponding maps
      *
      * @param data String containing the JSON-formatted response
      * @return List of maps with response properties
      */
     public static List<Map<String, String>> jsonStringToList(String data) {
+        FlowJsonObjectReader jsonReader = new FlowJsonObjectReader();
+        TypeReference<List<Map<String, String>>> typeReference = new TypeReference<List<Map<String, String>>>() {};
         try {
-            return JSON_OBJECT_MAPPER.readValue(data,
-                    new TypeReference<List<Map<String, String>>>() {});
+            return jsonReader.readObject(data, typeReference);
         } catch (IOException e) {
             // Data is not JSON-formatted
```

Comment:
I would recommend logging this to the console or this code may fail and we will never find out why, unless you are 100% sure the exception will never occur and then you should write a comment //ignored to show that we are awaire we are swallowing the exception and it is not something we forgot.

Reply:
I opted for ignoring the exceptions because the assertions will fail if the exception occurs anyway and we can figure out what happened